react/src/machine-coding/color-mixer/index.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from 'react';
+import './styles.module.css';
+
+function MixColors() {
+  const [color1, setColor1] = useState('#FF5733'); // Default color 1
+  const [color2, setColor2] = useState('#33FF57'); // Default color 2
+
+  const mixColors = () => {
+    // Mix the two colors
+    const r1 = parseInt(color1.slice(1, 3), 16);
+    const g1 = parseInt(color1.slice(3, 5), 16);
+    const b1 = parseInt(color1.slice(5, 7), 16);
+
+    const r2 = parseInt(color2.slice(1, 3), 16);
+    const g2 = parseInt(color2.slice(3, 5), 16);
+    const b2 = parseInt(color2.slice(5, 7), 16);
+
+    const mixedColor = `#${Math.round((r1 + r2) / 2).toString(16)}${Math.round((g1 + g2) / 2).toString(16)}${Math.round(
+      (b1 + b2) / 2
+    ).toString(16)}`;
+
+    return mixedColor;
+  };
```

Comment:
why can't move this function out of the component by passing the colors as arguments